perun-core/src/main/java/cz/metacentrum/perun/core/implApi/modules/attributes/UserAttributesModuleAbstract.java

```diff
@@ -35,4 +35,11 @@ public Attribute fillAttribute(PerunSessionImpl session, User user, AttributeDef
 	public void changedAttributeHook(PerunSessionImpl session, User user, Attribute attribute) throws WrongReferenceAttributeValueException {
 
 	}
+
+	public Attribute getAnonymizedValue(PerunSessionImpl session, User user, Attribute attribute) {
+		Attribute anonymizedAttribute = new Attribute(attribute);
+		anonymizedAttribute.setValue(null);
```

Comment:
I think 'null' is not appropriate as anonymized value of attribute, it may have unforeseen consequences later. I suggest using some hash and possibly check the attribute syntax too (is the situtation when you return value which is subsequently refused in setValue(), excluded?)

Reply:
Maybe we could use the current attribute value as default value instead of 'null'. What do you think @stavamichal ?

Reply:
The current value is the best option.

Reply:
I changed default value to the current value.

Reply:
Should not it be an error if you want to anonymize attribute that is not capable of anonymization?

Reply:
I feel that it is confusing, but what you want is to define a set of attributes you can't remove, and for these attributes, you want to ask Perun what you should set here as a value instead of the existing one. And if Perun will return the same value, it means there is no way how this value could be anonymized. Maybe the problem is in the name of the method, but the logic is right in my opinion.

Reply:
OK, then where is the logic that prevents deletion of these attributes if someone calls deleteUser()? 
Also the configuration suggests that the attributes should be anonymized, not kept untouched. 

Is there a way of saying "this attribute must be kept only in a form that enables comparison, but not reading the actual value"?

Reply:
We discussed this outside of this thread. The goal should be to create two arrays of processed attributes:
- attributes skipped by the removing/anonymization process
- attributes processed and anonymized (throwing an exception if there is no support in the module)